java/src/main/java/ai/rapids/cudf/Schema.java

```diff
@@ -194,6 +210,10 @@ public int[] getFlattenedTypeIds() {
     return ret;
   }
 
+  /**
+   * Get scales of the columns' types flattened from all levels in schema.
```

Comment:
nit: the key words I think we are missing are depth first.  Otherwise it is not clear how it was flattened.

Reply:
Updated, adding "depth-first traversal".